llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -129,10 +137,22 @@ def get_child_at_index(self, index):
 
         if not (0 <= index < self._num_children):
             return None
-        return self._evaluate(
-            "({self._element_type}){self._fn_prefix}_get({self._ptr}, {index})".format(self=self, index=index),
-            name=str(index)
-        )
+
+        if self._element_type == KREF:
+            self._evaluate(KREF + " $retval;")
+            return self._evaluate(
+                "({self._element_type}){self._fn_prefix}_get({self._ptr}, {index}, &$retval)".format(
+                    self=self, index=index
+                ),
+                name=str(index)
+            )
```

Comment:
I am not sure how well this branch of if interacts with runtime. Here, we synthesize a "local" variable slot in debugger, store a KObject in there, and increment some reference counts. Presumably, this leaks the object, because we never decrement the counter. I am not sure what to do here, and how is this supposed to work :) Related question: If I evaluate a C++ expression in C++, which gives me an `std::vector` back, is this vector destroyed eventually?

Reply:
Это не должно так работать - мы не должны держать явных ссылок на объекты, особенно в виде ObjHeader.